EngineLayer/CommonParameters.cs

```diff
@@ -2,133 +2,83 @@
-        public bool UseProvidedPrecursorInfo { get; private set; }
-        public double DeconvolutionIntensityRatio { get; private set; }
-        public int DeconvolutionMaxAssumedChargeState { get; private set; }
-        public Tolerance DeconvolutionMassTolerance { get; private set; }
-        public int TotalPartitions { get; private set; }
-        public bool BIons { get; private set; }
-        public bool YIons { get; private set; }
-        public bool ZdotIons { get; private set; }
-        public bool CIons { get; private set; }
-        public Tolerance ProductMassTolerance { get; private set; }
-        public Tolerance PrecursorMassTolerance { get; private set; }
-        public bool AddCompIons { get; private set; }
-        public double ScoreCutoff { get; private set; }
-        public DigestionParams DigestionParams { get; private set; }
-        public bool ReportAllAmbiguity { get; private set; }
-        public int TopNpeaks { get; private set; }
-        public double MinRatio { get; private set; }
-        public bool TrimMs1Peaks { get; private set; }
-        public bool TrimMsMsPeaks { get; private set; }
-        public bool UseDeltaScore { get; private set; }
-        public bool CalculateEValue { get; private set; }
+        // Search settings
 
-        public CommonParameters Clone()
-        {
-            return new CommonParameters(
-                bIons: this.BIons,
-                yIons: this.YIons,
-                zDotIons: this.ZdotIons,
-                cIons: this.CIons,
-                doPrecursorDeconvolution: this.DoPrecursorDeconvolution,
-                useProvidedPrecursorInfo: this.UseProvidedPrecursorInfo,
-                deconvolutionIntensityRatio: this.DeconvolutionIntensityRatio,
-                deconvolutionMaxAssumedChargeState: this.DeconvolutionMaxAssumedChargeState,
-                reportAllAmbiguity: this.ReportAllAmbiguity,
-                addCompIons: this.AddCompIons,
-                totalPartitions: this.TotalPartitions,
-                scoreCutoff: this.ScoreCutoff,
-                topNpeaks: this.TopNpeaks,
-                minRatio: this.MinRatio,
-                trimMs1Peaks: this.TrimMs1Peaks,
-                trimMsMsPeaks: this.TrimMsMsPeaks,
-                useDeltaScore: this.UseDeltaScore,
-                calculateEValue: this.CalculateEValue,
-                productMassTolerance: this.ProductMassTolerance,
-                precursorMassTolerance: this.PrecursorMassTolerance,
-                deconvolutionMassTolerance: this.DeconvolutionMassTolerance,
-                maxThreadsToUsePerFile: this.MaxThreadsToUsePerFile,
-                digestionParams: this.DigestionParams,
-                listOfModsVariable: this.ListOfModsVariable,
-                listOfModsFixed: this.ListOfModsFixed
-            );
-        }
+        public DigestionParams DigestionParams { get; set; } = new DigestionParams();
```

Comment:
I know from when I refactored Common Parameters a few months ago, eliminating the interface, that Rob wants the all the variables to be as secure as possible. This meaning only being set with the constructor or with a private set method(the initial goal was to have all variable be read only, but there are certain cases where the ability to set the variable outside of the constructor is necessary). You may want to discuss this with him further.

Reply:
yeah I really don't want these properties to be easily settable. IMO they should be only settable in the constructor (either readonly, or private with no set method). The settings should not change in the middle of a run and I want to prevent that from accidentally happening.